### PR TITLE
Unified handling of self-signed certificates

### PR DESCRIPTION
**Pull request by pontusmelke**
Bolt was ignoring user settings and always created a new self-signed
certificate on each boot. Changed so that server and bolt agree on the
same settings for certificates and private keys. Also stopped relying on
Nettys, `SelfSignedCertificate`, which are not safe to use in production.

### CODE REVIEW

**Comment by zhenlineo** [comment]
Would you mind to revert your final commit for the flaky tests, as this https://github.com/neo4j/neo4j/pull/6144 might be a better fix, WDYT?

**Comment by pontusmelke** [comment]
@zhenlineo I agree and have removed the fix

File: community/bolt/src/test/resources/certificates/combined.pem [hunk 1]
```diff
@@ -0,0 +1,71 @@
+-----BEGIN CERTIFICATE-----
```
Comment (zhenlineo) [inline community/bolt/src/test/resources/certificates/combined.pem:1]: I bet this would not be necessary. So does the chained_key.der?

Comment (pontusmelke) [inline community/bolt/src/test/resources/certificates/combined.pem:1]: I had to move the pem files we use in tests, since we moved the code

**Pull request merged**